### PR TITLE
Implement f{32,64}.abs instructions

### PR DESCRIPTION
**Pull request by gumb0**
image

### CODE REVIEW

File: lib/fizzy/execute.cpp [hunk 1]
```diff
@@ -1521,12 +1521,26 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, span<const Value>
             break;
         }
 
+        case Instr::f32_abs:
+        {
+            // TODO: This can be optimized https://godbolt.org/z/aPqvfo
```
Comment (gumb0) [inline lib/fizzy/execute.cpp:1526]: Included link to the same goldbolt as in #471

File: lib/fizzy/execute.cpp [hunk 2]
```diff
@@ -1521,12 +1521,26 @@ ExecutionResult execute(Instance& instance, FuncIdx func_idx, span<const Value>
...
+            unary_op(stack, static_cast<float (*)(float)>(std::fabs));
```
Comment (gumb0) [inline lib/fizzy/execute.cpp:1527]: `std::fabs` is unfortunately not a template, but several overloaded variants.

Comment (gumb0) [reply lib/fizzy/execute.cpp:1527]: Also would work without casts: `fabsf` and `fabs` C functions.

Comment (chfast) [reply lib/fizzy/execute.cpp:1527]: > (Also would work without casts: `fabsf` and `fabs` C functions.)

That would only work for `fabsf`...

File: test/unittests/execute_floating_point_test.cpp [hunk 3]
```diff
@@ -363,6 +385,27 @@ TYPED_TEST(execute_floating_point_types, add)
     EXPECT_THAT(exec(TypeParam{0x0.287p2}, TypeParam{0x1.FFp4}), Result(TypeParam{0x1.048Ep5}));
 }
 
+TYPED_TEST(execute_floating_point_types, abs)
```
Comment (chfast) [inline test/unittests/execute_floating_point_test.cpp:388]: Put the test by the opcode value, i.e. before `add`.

Comment (gumb0) [reply test/unittests/execute_floating_point_test.cpp]: Moved.

File: test/unittests/execute_floating_point_test.cpp [hunk 4]
```diff
@@ -293,6 +315,28 @@ TYPED_TEST(execute_floating_point_types, compare)
...
+    using FP = FP<TypeParam>;
+    using Limits = typename FP::Limits;
+
+    auto instance = instantiate(parse(this->get_unop_code(Instr::f32_abs)));
+    const auto exec = [&](auto arg) { return execute(*instance, 0, {arg}); };
+
+    std::vector p_values(
+        std::begin(this->positive_special_values), std::end(this->positive_special_values));
+    for (const auto x :
+        {TypeParam{0}, Limits::infinity(), FP::nan(FP::canon), FP::nan(FP::canon + 1), FP::nan(1)})
```
Comment (axic) [inline test/unittests/execute_floating_point_test.cpp:329]: Based on the spec, can you add `-0`?

**Pull request merged**